elasticsearch-dsl/spec/elasticsearch/dsl/search/queries/bool_spec.rb

```diff
@@ -32,6 +32,64 @@
 
   describe '#initialize' do
 
+    context 'when an object instance is provided' do
+
+      let(:search) do
+        described_class.new.must(Elasticsearch::DSL::Search::Queries::Match.new foo: 'bar')
+      end
+
+      it 'executes the block' do
+        expect(search.to_hash).to eq(bool: {must: [ {match: { foo: 'bar' }} ] })
+      end
```

Comment:
same here: I think the `#must` method is called on the `Elasticsearch::DSL::Search::Queries::Bool` and no block is executed.